src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1349,23 +1349,36 @@ abstract class BCodeBodyBuilder extends BCodeSkelBuilder {
       val constrainedType = MethodBType(lambdaParams.map(p => typeToBType(p.tpe)), typeToBType(lambdaTarget.tpe.resultType)).toASMType
       val samMethodType = methodBTypeFromSymbol(sam).toASMType
       val markers = if (addScalaSerializableMarker) classBTypeFromSymbol(definitions.SerializableClass).toASMType :: Nil else Nil
-      visitInvokeDynamicInsnLMF(bc.jmethod, sam.name.toString, invokedType, samMethodType, implMethodHandle, constrainedType, isSerializable, markers)
+      val overriddenMethods = enteringErasure(sam.allOverriddenSymbols).map(o => methodBTypeFromSymbol(o).toASMType)
```

Comment:
It would be ideal to unify the computation of the required bridges with the normal bridge logic in `Erasure`. We would need to change it to recognize that `Function` nodes are classes-in-waiting, and as such should have `ComputeBridges` applied. The listing of required bridges could be attached to the function symbol (probably inside the existing attachment, [`SAMFunction`](https://github.com/scala/scala/blob/f18b2494059599633f5afbefd57be8080ac21b0b/src/reflect/scala/reflect/internal/StdAttachments.scala#L57).

Reply:
The implementation of `ComputeBridges` as a `SymbolPair`-s walk has an efficiency argument over finding overriden members `decl`-wise. For SAM functions that won't make a difference, though, as we've only decl.

So the argument for unifying the logic would be more about keeping responsibilities clear, which might help curtail bugs down the track.